lib/team_statistics.rb

```diff
@@ -0,0 +1,200 @@
+  end
+
+  def best_season(passed_id)
+    best_and_worst_season_suite(passed_id)
+    @win_percentage_by_season.invert.max[1].to_s
+  end
+
+  def worst_season(passed_id)
+    best_and_worst_season_suite(passed_id)
+    @win_percentage_by_season.invert.min[1].to_s
+  end
+
+  def sort_games_won_by_team_id(passed_id)
+    @by_team_id_game_objects.each do |game|
+      if passed_id == game.away_team_id.to_s && game.away_goals > game.home_goals
+        @games_won_by_team_id[game.away_team_id.to_s] += 1
+      elsif passed_id == game.home_team_id.to_s && game.home_goals > game.away_goals
+        @games_won_by_team_id[game.home_team_id.to_s] += 1
+      elsif passed_id == game.away_team_id.to_s && game.away_goals < game.home_goals
+        @games_won_by_team_id[game.away_team_id.to_s] += 0
+      elsif passed_id == game.home_team_id.to_s && game.home_goals < game.away_goals
+        @games_won_by_team_id[game.home_team_id.to_s] += 0
+      end
+    end
+  end
+
+  def average_win_suite(passed_id)
+    collect_game_objects_by_team_id(passed_id)
+    @total_games_played = @by_team_id_game_objects.size
+    sort_games_won_by_team_id(passed_id)
+    @games_won_by_team = @games_won_by_team_id.values[0].to_f
+  end
+
+  def average_win_percentage(passed_id)
+    average_win_suite(passed_id)
+    (@games_won_by_team / @total_games_played).round(2)
+  end
+
+  def goals_by_game_by_team(passed_id)
+    @by_team_id_game_objects.each do |game|
+      if passed_id == game.away_team_id.to_s
+        @goals_by_game << game.away_goals
+      elsif passed_id == game.home_team_id.to_s
+        @goals_by_game << game.home_goals
+      end
+    end
+  end
+
+  def most_goals_scored(passed_id)
+    collect_game_objects_by_team_id(passed_id)
+    goals_by_game_by_team(passed_id)
+    @goals_by_game.max
+  end
+
+  def fewest_goals_scored(passed_id)
+    collect_game_objects_by_team_id(passed_id)
+    goals_by_game_by_team(passed_id)
+    @goals_by_game.min
+  end
+
+  def games_played_by_opponent_by_team(passed_id)
+    @by_team_id_game_objects.each do |game|
+      if passed_id == game.away_team_id.to_s
+        @games_played_by_opponent[game.home_team_id] += 1
+      elsif passed_id == game.home_team_id.to_s
+        @games_played_by_opponent[game.away_team_id] += 1
+      end
+    end
+  end
+
+  def games_won_by_opponent_by_team(passed_id)
+    @by_team_id_game_objects.each do |game|
+      if passed_id == game.away_team_id.to_s && game.away_goals > game.home_goals
+        @games_won_by_opponent[game.home_team_id] += 1
+      elsif passed_id == game.home_team_id.to_s && game.home_goals > game.away_goals
+        @games_won_by_opponent[game.away_team_id] += 1
+      elsif passed_id == game.away_team_id.to_s && game.away_goals < game.home_goals
+        @games_won_by_opponent[game.home_team_id] += 0
+      elsif passed_id == game.home_team_id.to_s && game.home_goals < game.away_goals
+        @games_won_by_opponent[game.away_team_id] += 0
+      end
+    end
```

Comment:
Lot of similar methods. Let's talk about how to implement a module

Reply:
I absolutely think we can pull out at least 2 methods from this class alone into a module. We need to check the other classes.